lib/cache.coffee

```diff
@@ -120,8 +120,15 @@ exports.getImageWritableStream = (deviceType, version) ->
 			# Ensure the cache directory exists, to prevent
 			# ENOENT errors when trying to write to it.
 			mkdirp(path.dirname(imagePath)).then ->
-				return fs.createWriteStream(imagePath)
+				# Append .inprogress to streams, move them to the right location only on success
+				inProgressPath = imagePath + '.inprogress'
```

Comment:
Would it be good to delete this file if the stream emits an `error` event?

Reply:
Great idea! Done, let me know what you think